src/main/java/klerer/citibike/lambda/StationsCache.java

```diff
@@ -0,0 +1,95 @@
+package klerer.citibike.lambda;
+
+import com.google.gson.Gson;
+import com.google.gson.stream.JsonReader;
+import klerer.citibike.CitiBikeService;
+import klerer.citibike.CitiBikeServiceFactory;
+import klerer.citibike.json.Stations;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.HeadObjectRequest;
+import software.amazon.awssdk.services.s3.model.HeadObjectResponse;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.time.Duration;
+import java.time.Instant;
+
+public class StationsCache {
+    //CHECKSTYLE:OFF
+    private final String BUCKET_NAME = "klerer.citibike";
+    private final String KEY_NAME = "station_information.json";
+    //CHECKSTYLE:ON
+    private Instant lastModified;
+    private Stations stations;
+    private Gson gson = new Gson();
+    CitiBikeService service;
+    S3Client s3Client;
+
+    public StationsCache() {
+        Region region = Region.US_EAST_2;
+        this.s3Client = S3Client.builder()
+                .region(region)
+                .credentialsProvider(DefaultCredentialsProvider.create())
+                .build();
+
+        this.service = new CitiBikeServiceFactory().getService();
+    }
+
+    public Stations getStations() {
+        boolean moreThanOneHour = s3Age();
+        if (stations != null && lastModified != null && moreThanOneHour) {
```

Comment:
Instead of checking the age directly from s3 here, check the age of the lastModified variable.